src/UserSpaceInstrumentation/InjectLibraryInTracee.cpp

```diff
@@ -185,21 +163,20 @@ ErrorMessageOr<uint64_t> FindFunctionAddressWithFallback(pid_t pid, std::string_
       .AppendBytes({0xff, 0xd0})
       .AppendBytes({0xcc});
 
-  auto return_value_or_error = ExecuteMachineCode(pid, address_code, memory_size, code);
-  if (return_value_or_error.has_error()) {
-    FreeMemoryOrDie(pid, address_code, memory_size);
-    return return_value_or_error.error();
-  }
-  return absl::bit_cast<void*>(return_value_or_error.value());
+  OUTCOME_TRY(return_value, ExecuteMachineCode(pid, address_code, code));
+
+  return absl::bit_cast<void*>(return_value);
 }
 
 [[nodiscard]] ErrorMessageOr<void> DlcloseInTracee(pid_t pid, void* handle) {
   // Figure out address of dlclose.
-  OUTCOME_TRY(address_dlclose, FindFunctionAddressWithFallback(pid, kDlcloseInLibdl, kLibdlSoname,
-                                                               kDlcloseInLibc, kLibcSoname));
+  OUTCOME_TRY(address_dlclose, FindFunctionAddressWithFallback(pid, kLibdlSoname, kDlcloseInLibdl,
+                                                               kLibcSoname, kDlcloseInLibc));
 
   // Allocate small memory area in the tracee.
-  OUTCOME_TRY(address_code, AllocateInTracee(pid, 0, kCodeScratchPadSize));
+  OUTCOME_TRY(address, AllocateInTracee(pid, 0, kCodeScratchPadSize));
+  orbit_base::unique_resource address_code{
+      address, [&pid](uint64_t address) { FreeMemoryOrDie(pid, address, kCodeScratchPadSize); }};
```

Comment:
maybe have AllocateInTracee return the `ErrorMessageOr<unique_resource<...>>` instead of raw pointer?

Reply:
Yes, this sounds good! There are a few places were this is used and I'm not entirely sure if there is something blocking this change so I'll defer that to a follow up PR. Tracked here: http://b/185315205